GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1803,6 +1803,11 @@ private void PopulateRecentRepositoriesMenu(ToolStripDropDownItem container)
 
             void AddRecentRepositories(Repository repo, string caption)
             {
+                if (AppSettings.RecentReposShowBranchName)
+                {
+                    caption = $"{caption} [{GitModule.GetSelectedBranchFast(repo.Path)}]";
+                }
```

Comment:
⚠️  
1. `GitModule.GetSelectedBranchFast` may return `string.Empty`, showing "caption []" won't look good.
2. The caption is modified below as well, it won't look good at all

Reply:
Will fix that it wont show just brackets without content

Reply:
GetSelectedBranchFast returns "(no branch)"/DetachedHeadParser.DetachedBranch if not set, string.Empty is not used (just open a GE subrepo where a branch is not checked out, the dafault to see this).
This should also be excluded.

Reply:
Done